File: task-3/prime.exs
defmodule PrimeChecker do
  # Function to check if a number is prime
  def is_prime(num) when num <= 1, do: false
  def is_prime(num) do
    Enum.all?((2..trunc(:math.sqrt(num))), fn i ->
      rem(num, i) != 0
    end)
  end

  # Main function
  def main do
    # Input from the user
    IO.write("Enter a number: ")
    n = String.trim(IO.gets(""))

    # Check if the number is prime
    case String.to_integer(n) do
      {:ok, number} ->
        if is_prime(number) do
          IO.puts("#{number} is a prime number.")
        else
          IO.puts("#{number} is not a prime number.")
        end
      _ ->
        IO.puts("Invalid input. Please enter a valid number.")
    end
  end
end

# Call the main function
PrimeChecker.main()

File: task-3/prime.cpp
#include<iostream>
using namespace std;

int main() {	
int n , c=0;
cout << "Enter a Number" << endl;
cin >> n ;
for(int i=2; i<=n;i++) {
if(i%n == 0) {
	c++;
}
}
if(c) {
cout << "Prime Number" <<endl;
}
else {
cout << "Not a Prime Number" << endl;
}
}
